### PR TITLE
Fix dicom wsi conversion

### PR DESCRIPTION
**Pull request by miriam-groeneveld**
Temporary fix until libvips 8.15 will be released, which will come with OpenSlide 4.0.0 which offers dicom-wsi support.

### CODE REVIEW

File: panimg/contrib/wsi_dcm_to_tiff/dcm_to_tiff.py [hunk 1]
```diff
@@ -228,46 +228,47 @@ def dcm_to_tiff(input_dir, output_path):
...
+        test_tile = np.array(image.read_tile(0, (0, 0)))
 
-        def tiler(getter, level, cols, rows):
-            for row in range(rows):
-                for col in range(cols):
-                    im = np.array(getter(level, (col, row)))
+        def tiler(getter, lvl, c, r):
+            for row in range(r):
+                for col in range(c):
+                    im = np.array(getter(lvl, (col, row)))
```
Comment (pkcakeout) [inline panimg/contrib/wsi_dcm_to_tiff/dcm_to_tiff.py:236]: Wasn't the old naming better?

**Review by pkcakeout — COMMENTED** [review]
Generally seems fine to me... not an expert on the matter though...

File: panimg/image_builders/tiff.py [hunk 2]
```diff
@@ -36,6 +36,13 @@
     pyvips = False
 
 
+DICOM_WSI_STORAGE_ID = "1.2.840.10008.5.1.4.1.1.77.1.6"
+
+
+def format_error(message: str) -> str:
+    return f"Tif image builder: {message}"
```
Comment (jmsmkn) [inline panimg/image_builders/tiff.py:43]:
```suggestion
    return f"TIFF image builder: {message}"
```

abbreviation of Tag Image File Format.

File: panimg/image_builders/tiff.py [hunk 3]
```diff
@@ -342,7 +349,11 @@ def _convert(
                 output_directory=output_directory,
             )
         except Exception as e:
-            file_errors[file].append(str(e))
+            file_errors[file].append(
+                format_error(
+                    f"Could not convert file to tif: {file.name}, error:{str(e)}"
```
Comment (jmsmkn) [inline panimg/image_builders/tiff.py:354]:
```suggestion
                    f"Could not convert file to TIFF: {file.name}, error:{str(e)}"
```

for consistency.

File: panimg/image_builders/tiff.py [hunk 4]
```diff
@@ -402,7 +413,11 @@ def _convert_dicom_wsi_dir(
 
         wsi_dcm_to_tiff(wsidicom_dir, new_file_name)
     except Exception as e:
-        file_errors[file].append(str(e))
+        file_errors[file].append(
+            format_error(
+                f"Could not convert dicom-wsi to tif: {file.name}, error:{str(e)}"
```
Comment (jmsmkn) [inline panimg/image_builders/tiff.py:418]:
```suggestion
                f"Could not convert dicom-wsi to TIFF: {file.name}, error:{str(e)}"
```

for consistency.

File: panimg/image_builders/tiff.py [hunk 5]
```diff
@@ -411,12 +426,61 @@ def _convert_dicom_wsi_dir(
...
+        if not headers:
+            continue
+
+        if not all(
+            header["data"].SOPClassUID == DICOM_WSI_STORAGE_ID
+            for header in headers
+        ):
+            for d in headers:
+                file_errors[d["file"]].append(
+                    format_error("Non-WSI-DICOM not supported by TIF builder")
```
Comment (jmsmkn) [inline panimg/image_builders/tiff.py:465]:
```suggestion
                    format_error("Non-WSI-DICOM not supported by TIFF builder")
```

for consistency.

File: tests/test_default_image_builders.py [hunk 6]
```diff
@@ -24,9 +24,10 @@
         "(supported formats: .fds,.fda,.e2e)"
     ],
     "image_builder_tiff": [
-        "Could not open file with tifffile.",
-        "Could not open file with OpenSlide.",
-        "Validation error: Not a valid tif: Image width could not be determined.",
+        "Tif image builder: Could not open file with tifffile.",
+        "Tif image builder: Could not open file with OpenSlide.",
+        "Tif image builder: Validation error: Not a valid tif: "
+        "Image width could not be determined.",
```
[the remark this answers is not in the document]
Comment (miriam-groeneveld) [reply tests/test_default_image_builders.py:30]: No, it's a single error

File: panimg/contrib/wsi_dcm_to_tiff/dcm_to_tiff.py [hunk 7]
```diff
@@ -211,11 +211,11 @@
...
 
 
-def is_dicom(dirpath):
+def is_dicom(dir_path):
     try:
-        path = Path(dirpath)
+        path = Path(dir_path)
         files = list(path.iterdir())
-        image = WsiDicom.open(files)
+        WsiDicom.open(files)
```
[the remark this answers is not in the document]
Comment (miriam-groeneveld) [reply panimg/contrib/wsi_dcm_to_tiff/dcm_to_tiff.py]: fixed

File: tests/test_default_image_builders.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jmsmkn) [reply tests/test_default_image_builders.py:30]: Yes but where does it come from?

[the remark this answers is not in the document]
Comment (jmsmkn) [reply tests/test_default_image_builders.py:30]: Oh wait, I see, line break.

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**